### PR TITLE
5. View a separate page for each category with a list of related reviews

### PR DESCRIPTION
**Pull request by JessalynKemp**
Only reviews related to category selected via nav-bar or on current URL path are rendered

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,18 +15,26 @@ function App() {
     name: "Guest Account",
     avatar_url: "https://cdn-icons-png.flaticon.com/512/1077/1077114.png?w=360",
   });
+  const [selectedCategory, setSelectedCategory] = useState(
+    window.location.pathname.substring(1) || "all"
```
Comment (karlaferry) [inline src/App.js]: Instead of doing this, you can use useParams :)

File: src/App.js [hunk 2]
```diff
@@ -14,18 +15,26 @@ function App() {
...
+          value={{ selectedCategory, setSelectedCategory }}
+        >
+          <div className="app"></div>
+          <header>
+            <Header />
+            <NavBar />
+          </header>
+          <Routes>
+            <Route path="/" element={<ReviewList />} />
+            <Route path="/:category" element={<ReviewList />} />
```
Comment (karlaferry) [inline src/App.js]: Here you have a parametric endpoint with `category` as the parameter. You can easily access this with a useParams hook :)

File: src/components/ReviewList.jsx [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
 const [reviews, setReviews] = useState([]);
+const {selectedCategory} = useContext(CategoryContext)
+
+let categoryTitleName = selectedCategory.split("-").join(" ")
+categoryTitleName = categoryTitleName[0].toUpperCase() + categoryTitleName.substring(1);
 
 useEffect(() => {
     axios.get("https://nc-games-jk.herokuapp.com/api/reviews").then(({data}) => {
-        setReviews(data.reviews);
+        const newReviews = data.reviews.filter((review) => {
```
Comment (karlaferry) [inline src/components/ReviewList.jsx]: We highly don't recommend fetching all the reviews and then filtering them. You have an endpoint in your API that allows to query reviews by category, so make use of that :)

**Review by karlaferry — CHANGES_REQUESTED** [review]
Just some tiny changes. :) Please send new PR after :) Thank you!

**Comment by JessalynKemp** [comment]
Thanks Karla - not sure why I did that in such a roundabout way... 😆

**Review by karlaferry — APPROVED** [review]
Awesome, that's more like it 😂 ❤️

**Pull request merged**